internal-messaging/unixdomain/cc/UnixDomainStreamSocketClient.cc

```diff
@@ -35,22 +35,29 @@ int UnixDomainStreamSocketClient::Send(char message[]) {
     }
     return 0;
 }
-
-int UnixDomainStreamSocketClient::SendMessageAwaitReply(char message[]) {
+//TODO Should I return a string or should I pass in a char array pointer...
+string UnixDomainStreamSocketClient::SendMessageAwaitReply(char message[]) {
     if (WriteToSocket(message, socket_file_descriptor_) != 0) {
         error("ERROR SENDING MESSAGE");
-        return 1;
+        return "FAILURE";
```

Comment:
I feel like the reply should be stored in a separate pointer with this function returning true or false. Returning strings like this leads to unnecessary complexity